library/memory_buffer_alloc.c

```diff
@@ -247,6 +249,8 @@ static void *buffer_alloc_calloc( size_t n, size_t size )
 
     if( n != 0 && len / n != size )
```

Comment:
@andresag01 - I read the bug differently. That's because `calloc()` can return `NULL` or non-`NULL` values, we need to be able to handle both because we can't always depend on the underlying implementation to do what we expect. If, as you say, "some parts of the library assume that allocating 0 will return a non-NULL value", surely we need to fix this to accommodate both types of behaviour?

@andresag01 - I think you'd best raise a new bug to address other parts of the library.

I also think we need to treat a size 0 allocation as NULL.